packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -408,8 +412,14 @@ module.exports = (
       },
       tableOfContents: {
         type: GraphQLString,
-        resolve(markdownNode) {
-          return getTableOfContents(markdownNode)
+        args: {
+          field: {
+            type: GraphQLString,
+            defaultValue: `slug`,
```

Comment:
How about:
- renaming `field` to `pathToSlugField`
- changing `defaultValue` to `fields.slug`
- changing `markdownNode.fields[field]` to `_.get(markdownNode, pathToSlugField)`
?

This would allow more flexibility - if someone define slugs in frontmatter - he could set path to field as `frontmatter.slug` or `frontmatter.path` and don't require to create node field duplicate

Reply:
yeah definitely i haven't thought about using frontmatter, thanks!

Reply:
i dont know what process you follow, let me know if i have to squash commits or fixup or.. :)